org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -194,6 +197,17 @@ public Charset getCharset() {
     public <G> G getClientObject() {
       return (G) file;
     }
+
+    @Override
+    public String contents() throws IOException {
+      return new String(Files.readAllBytes(filePath), getCharset());
+
+    }
+
+    @Override
+    public InputStream inputStream() throws IOException {
+      return Files.newInputStream(filePath);
```

Comment:
Hm, that looks much better than the way I did it... (My head was full of IResource stuff...)